src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -92,14 +92,20 @@ const DomainFragment = ({
 
     navigation.push('DomainScreen', navigationParam);
   };
+  const handleScroll = React.useCallback(() => {
+    Keyboard.dismiss();
+  });
 
   const handleDomain = async (from, willFollow, item, index) => {
     if (from === FROM_FOLLOWED_DOMAIN_INITIAL) {
       const newFollowedDomains = [...domains];
-      newFollowedDomains[index].following = !!willFollow;
-      newFollowedDomains[index].user_id_follower = myId;
+      const domainIndex = newFollowedDomains.findIndex(
+        (domain) => domain.domain_page_id === item.domain_page_id
+      );
+      newFollowedDomains[domainIndex].following = !!willFollow;
+      // newFollowedDomains[domainIndex].user_id_follower = myId;
 
-      const newDomain = newFollowedDomains[index];
+      const newDomain = newFollowedDomains[domainIndex];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic to find the domain index based on `domain_page_id` instead of using the array index is a good improvement. This makes the code more robust against changes in the order or number of elements in the `domains` array.
```diff
- newFollowedDomains[index].following = !!willFollow;
- newFollowedDomains[index].user_id_follower = myId;
- const newDomain = newFollowedDomains[index];
+ const domainIndex = newFollowedDomains.findIndex(
+   (domain) => domain.domain_page_id === item.domain_page_id
+ );
+ newFollowedDomains[domainIndex].following = !!willFollow;
+ // newFollowedDomains[domainIndex].user_id_follower = myId;
+ const newDomain = newFollowedDomains[domainIndex];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->